go/backend/store/cache/cachedstore.go

```diff
@@ -75,10 +76,8 @@ func (m *Store[I, V]) Flush() error {
 }
 
 func (m *Store[I, V]) Close() error {
-	if err := m.Flush(); err != nil {
-		return err
-	}
-	return m.store.Close()
+	m.cache.Clear()
+	return errors.Join(m.Flush(), m.store.Close())
```

Comment:
Shouldnt we flush before clearing the cache?

Reply:
in this case, flush does not use the cache. But you are right, considering this as a black-box, it is correct to flush first. Fixed.